### PR TITLE
📝 Rename Websocket by WebSocket

### PR DESCRIPTION
**Pull request by Kludex**
This is fundamentally unnecessary, so the rejecting this PR is understandable. But the correct name is WebSocket.

### CODE REVIEW

**Comment by andrewgodwin** [comment]
I don't object to it, but if you want to land this you'll need to do it in a backwards-compatible fashion (so the old names work but raise a warning)

**Comment by Kludex** [comment]
I've used [PEP 562](https://www.python.org/dev/peps/pep-0562/) idea here, but as it was introduced in 3.7, I've used the [backport implementation for 3.6](https://github.com/facelessuser/pep562).

I've also added tests to check that the warning is being raised on the wanted types.

**Comment by andrewgodwin** [comment]
I'd rather just do it the "old fashioned" way if we could - expose them as actual top-level module objects rather than bringing in extra code just for this.

**Comment by Kludex** [comment]
I'm not following you. :thinking: How am I supposed to warn the user about the deprecation at import time without this implementation?

Or you just meant to not use `__all__` and `__deprecated__`?

**Comment by andrewgodwin** [comment]
Oh, I forgot they were type classes for a second, sorry. Thought they were actual classes where you could do it with a mixin on instantiation.

In that case, let's vendor pep562 in provided the license matches - I don't want more dependencies!

File: asgiref/_pep562.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    def __init__(self, name: str) -> None:
+        """Acquire `__getattr__` and `__dir__`, but only replace module for versions less than Python 3.7."""
+
+        self._module = sys.modules[name]
+        self._get_attr = getattr(self._module, "__getattr__", None)
+        self._get_dir: Optional[Callable[..., List[str]]] = getattr(
+            self._module, "__dir__", None
+        )
+        sys.modules[name] = self  # type: ignore[assignment]
```
Comment (Kludex) [inline asgiref/_pep562.py:37]: `sys.modules` expects a `Module` and `self` is a `Pep562`, jfyk.

**Comment by Kludex** [comment]
There's a mismatch between the linters, `isort` and `black` are reformatting themselves on the `pre-commit` :thinking:

**Comment by Kludex** [comment]
Notes about the changes: 
- I've added `pep562` as private module, so we can remove easily by the [3.6 EOL](https://endoflife.date/python).
- I've removed all the unnecessary code on `_pep562.py` (and tried to modify the less I could, jfyk).

**Comment by andrewgodwin** [comment]
I think sticking with `black`'s default length of 88 is the best approach - let's handle that in a separate PR. I'll merge this in now so we can avoid mixing linting in here - I think it looks good now, so thanks for bearing with my requests!

**Comment by Kludex** [comment]
Thank you! :tada:

**Comment by Kludex** [comment]
This is not `mypy` compliant. A project with `asgiref` not pinned, will receive this on the upgrade:
```python
+ mypy --show-error-codes
uvicorn/middleware/proxy_headers.py:13: error: Module "asgiref.typing" has no attribute "WebsocketScope"; maybe "WebSocketScope"?  [attr-defined]
```

Which I think it's expected. but I'm mentioning because I'm not sure you were aware. @andrewgodwin

**Pull request merged**